src/login/logind-inhibit.c

```diff
@@ -485,29 +485,34 @@ const char *inhibit_what_to_string(InhibitWhat w) {
 
 InhibitWhat inhibit_what_from_string(const char *s) {
         InhibitWhat what = 0;
-        const char *word, *state;
-        size_t l;
 
-        FOREACH_WORD_SEPARATOR(word, l, s, ":", state) {
-                if (l == 8 && strneq(word, "shutdown", l))
+        for (const char *p = s;;) {
+                _cleanup_free_ char *word = NULL;
+                int r;
+
+                r = extract_first_word(&p, &word, ":", EXTRACT_DONT_COALESCE_SEPARATORS);
+                if (r < 0)
+                        return r;
```

Comment:
I added an assert_cc that the value is fine. I think this is good enough — the compiler deciding to make the type smaller than a word would be very strange.

Reply:
Well sure, if it decides, the code will fail to compile. If you absolutely require this, I'll change it, but our code is very unlikely to be compiled by anything except gcc or clang (because there's just too many extensions that are not supported elsewhere), and both gcc and clang are very unlikely to start doing something different here.

Reply:
I still think this should be changed. I wonder if coverity or so would be tripped up by this the instant this is merged...

Also not that the choice of word size might depend on arch, and the casts here are implicit, i.e. the compiler might assume the user has checked the value fits into the type, and weird stuff might happen if that's not the case...

Reply:
I changed the return type to `int` now.